src/TRoom.cpp

```diff
@@ -237,23 +238,19 @@ void TRoom::setWeight(int w)
         w = 1;
     }
     weight = w;
+    mpRoomDB->mpMap->mUnsavedMap = true;
 }
 
-// Previous implementations did not allow for REMOVAL of an exit weight (by
-// setting it to zero)
 void TRoom::setExitWeight(const QString& cmd, int w)
 {
     if (w > 0) {
         exitWeights[cmd] = w;
-        if (mpRoomDB && mpRoomDB->mpMap) {
-            mpRoomDB->mpMap->mMapGraphNeedsUpdate = true;
-        }
+        mpRoomDB->mpMap->mMapGraphNeedsUpdate = true;
     } else if (exitWeights.contains(cmd)) {
         exitWeights.remove(cmd);
-        if (mpRoomDB && mpRoomDB->mpMap) {
-            mpRoomDB->mpMap->mMapGraphNeedsUpdate = true;
-        }
+        mpRoomDB->mpMap->mMapGraphNeedsUpdate = true;
     }
+    mpRoomDB->mpMap->mUnsavedMap = true;
```

Comment:
Needs to be moved up into each of the `if ()` and `else if () {` code blocks. So setting/removing an exit weight for a non-existent exit does not dirty the map. OTOH the validity for *setting* an exit weight is not having the given exit being checked here... 😟